types/helper.go

```diff
@@ -22,25 +22,48 @@ import (
 	"github.com/pingcap/errors"
 )
 
+// RoundRule indicate the 2 round rule for The ROUND() function
+type RoundRule int
+
+const (
+	// RoundHalfUp For exact-value numbers, ROUND() uses the “round half up” rule
+	RoundHalfUp RoundRule = iota
+
+	// RoundNearestEven For approximate-value numbers, the result depends on the C library. On many systems,
+	// this means that ROUND() uses the “round to nearest even” rule
+	RoundNearestEven
+)
+
 // RoundFloat rounds float val to the nearest even integer value with float64 format, like MySQL Round function.
-// RoundFloat uses default rounding mode, see https://dev.mysql.com/doc/refman/5.7/en/precision-math-rounding.html
-// so rounding use "round to nearest even".
+// see https://dev.mysql.com/doc/refman/5.7/en/precision-math-rounding.html
 // e.g, 1.5 -> 2, -1.5 -> -2.
 func RoundFloat(f float64) float64 {
 	return math.RoundToEven(f)
 }
 
-// Round rounds the argument f to dec decimal places.
+// Round rounds the argument f to dec decimal places use “round to nearest even” rule as default.
 // dec defaults to 0 if not specified. dec can be negative
 // to cause dec digits left of the decimal point of the
 // value f to become zero.
-func Round(f float64, dec int) float64 {
+func Round(f float64, dec int, r ...RoundRule) float64 {
```

Comment:
It looks too hacky to me, perhaps we can just add a `rule` parameter to the signature of this function and change all the places where it is used.

Reply:
Maybe we should add a function `func RoundInt64(f int64, dec int) int64`?

Reply:
yes, i came up with 2 change way:
- current one
   * prons: less change, adhere to golang which is only [float64 version of round 
(https://golang.google.cn/pkg/math/#Round) also; and since it is `float64`, it make sense use  `RoundNearestEven` rule as default
   * crons: a little performance affact
- addtional methons like `RoundInt64`
   * prons: better performance, straightforward, maybe adhere to mysql implements better
   * more change

Reply:
if you agree later one is better, i can update

Reply:
I think it's better that integer round does not reuse floating-point round, because `Float64` can't represent all integer values and will introduce inaccuracy in computation.

I can give an example that shows why floating-point round isn't suitable:

In TiDB:

```
mysql> select round(123456789, -5);
+----------------------+
| round(123456789, -5) |
+----------------------+
|            123499999 |
+----------------------+
1 row in set (0.001 sec)
```

In MySQL 8.0:

```
mysql> select round(123456789, -5);
+----------------------+
| round(123456789, -5) |
+----------------------+
|            123500000 |
+----------------------+
1 row in set (0.006 sec)
```

TiDB's output is expected for floating-point round even if you use "round to nearest" rule: <https://coliru.stacked-crooked.com/a/a3dbb35ff61b3944>.

Reply:
By the way, MySQL 8 will report an error if round result overflows:

```
mysql> select round(18446744073709551615, -19);
ERROR 1690 (22003): BIGINT UNSIGNED value is out of range in 'round(18446744073709551615,-(19))'
```

Reply:
thank you for suggestion, i will add a round function for integer, and adjust some old names to make them consistent

Reply:
> can be fixed by use convert `math.Round(f)`

I can offer another example:

TiDB:

```
mysql> select round(2146213728964879326, -15);
+---------------------------------+
| round(2146213728964879326, -15) |
+---------------------------------+
|             2145999999999999744 |
+---------------------------------+
1 row in set (0.000 sec)
```

MySQL 8.0:

```
mysql> select round(2146213728964879326, -15);
+---------------------------------+
| round(2146213728964879326, -15) |
+---------------------------------+
|             2146000000000000000 |
+---------------------------------+
1 row in set (0.001 sec)
```

I guess this one cannot be simply fixed by `math.Round`.

Reply:
yes, you are right, round seems not as easy as thought, see the link in my comments.
I will add your example as test case and hopefully get a pefect solution.